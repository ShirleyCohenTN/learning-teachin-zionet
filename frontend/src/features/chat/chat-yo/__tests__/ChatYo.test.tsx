import { describe, it, expect, vi } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import { ChatYo } from "../index";

// Mocking the hook used in ChatYo
vi.mock("../hooks", () => {
  return {
    useChat: () => ({
      sendMessage: vi.fn(),
      loading: false,
      messages: [
        {
          text: "Hello",
          position: "right",
          date: new Date(),
        },
        {
          text: "Hi there!",
          position: "left",
          date: new Date(),
        },
      ],
    }),
  };
});

// Snapshot and basic interaction test
describe("ChatYo", () => {
  it("renders chat messages correctly", () => {
    render(<ChatYo />);
    expect(screen.getByText("Hello")).toBeInTheDocument();
    expect(screen.getByText("Hi there!")).toBeInTheDocument();
  });

  it("allows typing and sending a message", () => {
    render(<ChatYo />);
    const input = screen.getByPlaceholderText("Type a message...");
    fireEvent.change(input, { target: { value: "Test message" } });
    expect(input).toHaveValue("Test message");

    const button = screen.getByRole("button", { name: "â†‘" });
    fireEvent.click(button);
    expect(input).toHaveValue(""); // input cleared after sending
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<ChatYo />);
    expect(asFragment()).toMatchSnapshot();
  });
});
