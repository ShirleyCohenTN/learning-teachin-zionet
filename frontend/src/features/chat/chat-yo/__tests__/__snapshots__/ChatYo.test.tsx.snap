// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChatYo > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="chatWrapper-0-2-1"
  >
    <div
      class="messagesList-0-2-2"
    >
      <div
        class="rce-container-mbox messageBox-0-2-3"
      >
        <div
          class="rce-mbox rce-mbox-right"
          style="background-color: rgb(17, 187, 255); color: rgb(0, 0, 0);"
        >
          <div
            class="rce-mbox-body"
          >
            <div
              class="rce-mbox-forward rce-mbox-forward-left"
            >
              <svg
                class=""
                color="currentColor"
                fill="none"
                height="20"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.7109 9.3871C21.8404 9.895 21.9249 10.4215 21.9598 10.9621C22.0134 11.7929 22.0134 12.6533 21.9598 13.4842C21.6856 17.7299 18.3536 21.1118 14.1706 21.3901C12.7435 21.485 11.2536 21.4848 9.8294 21.3901C9.33896 21.3574 8.8044 21.2403 8.34401 21.0505C7.83177 20.8394 7.5756 20.7338 7.44544 20.7498C7.31527 20.7659 7.1264 20.9052 6.74868 21.184C6.08268 21.6755 5.24367 22.0285 3.99943 21.9982C3.37026 21.9829 3.05568 21.9752 2.91484 21.7349C2.77401 21.4946 2.94941 21.1619 3.30021 20.4966C3.78674 19.5739 4.09501 18.5176 3.62791 17.6712C2.82343 16.4623 2.1401 15.0305 2.04024 13.4842C1.98659 12.6533 1.98659 11.7929 2.04024 10.9621C2.31441 6.71638 5.64639 3.33448 9.8294 3.05621C10.2156 3.03051 10.6067 3.01177 11 3"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
                <path
                  d="M8.5 15H15.5M8.5 10H12"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
                <path
                  d="M22 4.5L14 4.5M22 4.5C22 3.79977 20.0057 2.49153 19.5 2M22 4.5C22 5.20023 20.0057 6.50847 19.5 7"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </div>
            <div
              class="rce-mbox-forward rce-mbox-reply-btn-left"
            >
              <svg
                class=""
                color="currentColor"
                fill="none"
                height="20"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19.5 5.5L18.8803 15.5251C18.7219 18.0864 18.6428 19.3671 18.0008 20.2879C17.6833 20.7431 17.2747 21.1273 16.8007 21.416C15.8421 22 14.559 22 11.9927 22C9.42312 22 8.1383 22 7.17905 21.4149C6.7048 21.1257 6.296 20.7408 5.97868 20.2848C5.33688 19.3626 5.25945 18.0801 5.10461 15.5152L4.5 5.5"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-width="1.5"
                />
                <path
                  d="M3 5.5H21M16.0557 5.5L15.3731 4.09173C14.9196 3.15626 14.6928 2.68852 14.3017 2.39681C14.215 2.3321 14.1231 2.27454 14.027 2.2247C13.5939 2 13.0741 2 12.0345 2C10.9688 2 10.436 2 9.99568 2.23412C9.8981 2.28601 9.80498 2.3459 9.71729 2.41317C9.32164 2.7167 9.10063 3.20155 8.65861 4.17126L8.05292 5.5"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-width="1.5"
                />
                <path
                  d="M9.5 16.5L9.5 10.5"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-width="1.5"
                />
                <path
                  d="M14.5 16.5L14.5 10.5"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-width="1.5"
                />
              </svg>
            </div>
            <div
              class="rce-mbox-title rce-mbox-title--clear"
              style="color: black;"
            >
              <span>
                Me
              </span>
            </div>
            <div
              class="rce-mbox-text right"
            >
              Hello
            </div>
            <div
              class="rce-mbox-time non-copiable"
              data-text="just now"
            >
              <span
                class="rce-mbox-status"
              >
                <svg
                  class=""
                  color="currentColor"
                  fill="none"
                  height="14"
                  viewBox="0 0 24 24"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2.5 13.8333L6 17.5L7.02402 16.4272M16.5 6.5L10.437 12.8517"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="1.5"
                  />
                  <path
                    d="M7.5 13.8333L11 17.5L21.5 6.5"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="1.5"
                  />
                </svg>
              </span>
            </div>
          </div>
          <svg
            class="rce-mbox-right-notch"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0v20L20 0"
            />
          </svg>
        </div>
      </div>
      <div
        class="rce-container-mbox messageBox-0-2-3"
      >
        <div
          class="rce-mbox"
          style="background-color: rgb(255, 255, 255); color: rgb(0, 0, 0);"
        >
          <div
            class="rce-mbox-body"
          >
            <div
              class="rce-mbox-forward rce-mbox-forward-right"
            >
              <svg
                class=""
                color="currentColor"
                fill="none"
                height="20"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.7109 9.3871C21.8404 9.895 21.9249 10.4215 21.9598 10.9621C22.0134 11.7929 22.0134 12.6533 21.9598 13.4842C21.6856 17.7299 18.3536 21.1118 14.1706 21.3901C12.7435 21.485 11.2536 21.4848 9.8294 21.3901C9.33896 21.3574 8.8044 21.2403 8.34401 21.0505C7.83177 20.8394 7.5756 20.7338 7.44544 20.7498C7.31527 20.7659 7.1264 20.9052 6.74868 21.184C6.08268 21.6755 5.24367 22.0285 3.99943 21.9982C3.37026 21.9829 3.05568 21.9752 2.91484 21.7349C2.77401 21.4946 2.94941 21.1619 3.30021 20.4966C3.78674 19.5739 4.09501 18.5176 3.62791 17.6712C2.82343 16.4623 2.1401 15.0305 2.04024 13.4842C1.98659 12.6533 1.98659 11.7929 2.04024 10.9621C2.31441 6.71638 5.64639 3.33448 9.8294 3.05621C10.2156 3.03051 10.6067 3.01177 11 3"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
                <path
                  d="M8.5 15H15.5M8.5 10H12"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
                <path
                  d="M22 4.5L14 4.5M22 4.5C22 3.79977 20.0057 2.49153 19.5 2M22 4.5C22 5.20023 20.0057 6.50847 19.5 7"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </div>
            <div
              class="rce-mbox-forward rce-mbox-reply-btn-right"
            >
              <svg
                class=""
                color="currentColor"
                fill="none"
                height="20"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19.5 5.5L18.8803 15.5251C18.7219 18.0864 18.6428 19.3671 18.0008 20.2879C17.6833 20.7431 17.2747 21.1273 16.8007 21.416C15.8421 22 14.559 22 11.9927 22C9.42312 22 8.1383 22 7.17905 21.4149C6.7048 21.1257 6.296 20.7408 5.97868 20.2848C5.33688 19.3626 5.25945 18.0801 5.10461 15.5152L4.5 5.5"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-width="1.5"
                />
                <path
                  d="M3 5.5H21M16.0557 5.5L15.3731 4.09173C14.9196 3.15626 14.6928 2.68852 14.3017 2.39681C14.215 2.3321 14.1231 2.27454 14.027 2.2247C13.5939 2 13.0741 2 12.0345 2C10.9688 2 10.436 2 9.99568 2.23412C9.8981 2.28601 9.80498 2.3459 9.71729 2.41317C9.32164 2.7167 9.10063 3.20155 8.65861 4.17126L8.05292 5.5"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-width="1.5"
                />
                <path
                  d="M9.5 16.5L9.5 10.5"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-width="1.5"
                />
                <path
                  d="M14.5 16.5L14.5 10.5"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-width="1.5"
                />
              </svg>
            </div>
            <div
              class="rce-mbox-title rce-mbox-title--clear"
              style="color: gray;"
            >
              <div
                class="rce-avatar-container default default"
              >
                <img
                  class="rce-avatar"
                  src="/src/features/chat/chat-yo/assets/avatar1.png"
                />
              </div>
              <span>
                Assistant
              </span>
            </div>
            <div
              class="rce-mbox-text left"
            >
              Hi there!
            </div>
            <div
              class="rce-mbox-time non-copiable"
              data-text="just now"
            >
              <span
                class="rce-mbox-status"
              >
                <svg
                  class=""
                  color="currentColor"
                  fill="none"
                  height="14"
                  viewBox="0 0 24 24"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2.5 13.8333L6 17.5L7.02402 16.4272M16.5 6.5L10.437 12.8517"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="1.5"
                  />
                  <path
                    d="M7.5 13.8333L11 17.5L21.5 6.5"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="1.5"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div>
            <svg
              class="rce-mbox-left-notch"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <filter
                  id="filter1"
                  x="0"
                  y="0"
                >
                  <feoffset
                    dx="-2"
                    dy="-5"
                    in="SourceAlpha"
                    result="offOut"
                  />
                  <fegaussianblur
                    in="offOut"
                    result="blurOut"
                    stdDeviation="3"
                  />
                  <feblend
                    in="SourceGraphic"
                    in2="blurOut"
                    mode="normal"
                  />
                </filter>
              </defs>
              <path
                d="M20 0v20L0 0"
                filter="url(#filter1)"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rce-container-input input-0-2-6"
    >
      <input
        class="rce-input"
        placeholder="Type a message..."
        type="text"
        value=""
      />
      <div
        class="rce-input-buttons"
      >
        <button
          class="sendButton-0-2-7"
        >
          ↑
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
